### PR TITLE
Updated default and backup STT geometry for SAND added.

### PR DESCRIPTION
**Pull request by nibir98**
The updated, default and backup geometries are added as a standalone build.

### CODE REVIEW

File: build_hall.sh [hunk 1]
```diff
@@ -443,6 +443,36 @@ gegede-cli duneggd/Config/WORLDggd.cfg \
...
+if [ $option = "all" -o $option = "only_sand_def_stt" ];
+then
+gegede-cli duneggd/Config/WORLDggd.cfg \
+           duneggd/Config/ND_Hall_Air_Volume_Only_SAND.cfg \
+           duneggd/Config/ND_Hall_Rock.cfg \
+           duneggd/Config/ND_ElevatorStruct.cfg \
+           duneggd/Config/SAND_MAGNET.cfg \
+           duneggd/Config/SAND_INNERVOLOPT_DefaultSTT.cfg \
+           duneggd/Config/SAND_ECAL.cfg \
+           duneggd/Config/SAND_STT/STT_Default.cfg \
```
Comment (ast0815) [inline build_hall.sh:456]: You have this "default geometry", but it is not actually used in the default that is built when no argument is given to "build_hall.sh". If this is the default, then it should be used for the "prod" options. Currently, that is still using "STT1.cfg", whatever that is.

File: duneggd/Config/SAND_STT/STT_Backup.cfg [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+nofDownstreamTrkMod     = 6
+nofC3H6ModAfterCMod     = 6
+nofC3H6ModNoRadAfterCMod = 3
+
+#**********************************|   STT MODULE    |********************************************
+
+# target
+FrameThickness          = Q("10cm")
+AddGapForSlab           = Q("7cm")
+#targetThickness         = {"CMod" : Q("4mm"),    "C3H6Mod": Q("5mm"), "C3H6ModNoRad": Q("6.89mm"), "TrkMod":Q("0mm"), "TrkMod2lyr":Q("0mm")}
```
Comment (ast0815) [inline duneggd/Config/SAND_STT/STT_Backup.cfg:28]: What is with these commented out configurations? We are using version control, so unless there is a very good reason, unused code should just be removed and not be left in as comments. Especially if there are no explanation in the comments why it is commented out.

Same in multiple places in the code.

File: duneggd/SubDetector/SandInnerVolume.py [hunk 3]
```diff
@@ -28,9 +28,13 @@ def build_tracker(self, main_lv, geom):
         # if "STT" not in self.builders:
         #     print("STT builder not found")
         #     return  
+        #print(self.builders)
```
Comment (ast0815) [inline duneggd/SubDetector/SandInnerVolume.py:31]: See above comment about commented out code.

**Review by ast0815 — CHANGES_REQUESTED** [review]

File: build_hall.sh (hunk 1 above)
Comment (nibir98) [reply build_hall.sh:456]: The default configuration, has some volume naming changes with some slight modifications. The sandreco package is sensitive to volume names. In order to not break the sandreco code due to volume name conflicts, this is not added as part of the prod geometries as we talked about sometime ago. Moreover after discussion we have decided to rename the argument in the build_hall.sh script as default -> complete, backup -> initial respectively.

File: duneggd/Config/SAND_STT/STT_Backup.cfg (hunk 2 above)
Comment (nibir98) [reply duneggd/Config/SAND_STT/STT_Backup.cfg:28]: The commented section is cleared now.

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to a modified version of [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 Geometry releases will be tagged as `Descriptive_tag_v_X.Y.Z`.
 
+### Added
+
+- Updated STT geometry with Default and Backup options. Can be built as a standalone geometry. `build_hall.sh only_sand_def_stt` `build_hall.sh only_sand_bck_stt`
```
Comment (ast0815) [inline CHANGELOG.md:11]:
```suggestion
- Added STT geometry with Complete and Initial options. Can be built as a standalone geometry: `build_hall.sh only_sand_def_stt` `build_hall.sh only_sand_initial_stt`. Not yet part of the default "production" detector hall.
```

**Review by ast0815 — CHANGES_REQUESTED** [review]
Ok, this looks fine to me. I just suggested a change to the changelog entry. Please check if this is acceptable.

Just be aware, that since your geometry changes are not part of the "production" detector hall geometry, they have _not_ been tested by the CI for overlaps. You have to do that yourself with the included scripts.

When you have modified the changelog entry and are certain the geometry is correct, I am happy to merge this.

**Comment by ast0815** [comment]
I have changed the way we build the default geometries. They are now defined in a Makefile rather than a bash script. Could you pull those changes from `master` and add your new geometries as a target there?

File: duneggd/SubDetector/SandInnerVolume.py (hunk 3 above)
Comment (nibir98) [reply duneggd/SubDetector/SandInnerVolume.py]: this has been cleared.

File: CHANGELOG.md (hunk 4 above)
Comment (nibir98) [reply CHANGELOG.md:11]: Updated the log file.

Comment (nibir98) [reply CHANGELOG.md:11]: Updated to the makefile method from the build_hall.sh method

File: CHANGELOG.md [hunk 5]
```diff
@@ -8,10 +8,11 @@ Geometry releases will be tagged as `Descriptive_tag_v_X.Y.Z`.
 
 ### Added
 
-- Updated STT geometry with Default and Backup options. Can be built as a standalone geometry. `build_hall.sh only_sand_def_stt` `build_hall.sh only_sand_bck_stt`
-
 ## [Unreleased]
 
+### Added
+- Updated with standalone build of initial and complete STT configurations. `only_SAND_STT_Initial.gdml` `only_SAND_STT_Complete.gdml`
```
Comment (ast0815) [inline CHANGELOG.md:14]: There is already an 'Added' subsection below. Please just add your line there.

File: Makefile [hunk 6]
```diff
@@ -201,6 +203,27 @@ only_SAND_DRIFT.gdml: duneggd/Config/WORLDggd.cfg \
...
+		   duneggd/Config/ND_Hall_Air_Volume_Only_SAND.cfg \
+		   duneggd/Config/ND_Hall_Rock.cfg \
+		   duneggd/Config/ND_ElevatorStruct.cfg \
+		   duneggd/Config/SAND_MAGNET.cfg \
+		   duneggd/Config/SAND_INNERVOLOPT_DefaultSTT.cfg \
+		   duneggd/Config/SAND_ECAL.cfg \
+		   duneggd/Config/SAND_STT/STT_Default.cfg \
+		   duneggd/Config/SAND_GRAIN.cfg \
+
+
```
Comment (ast0815) [inline Makefile:226]: Could you please follow the indentation convention in this file and use a single tab character?

**Review by ast0815 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to a modified version of [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 Geometry releases will be tagged as `Descriptive_tag_v_X.Y.Z`.
 
+### Added
+
```
Comment (ast0815) [inline CHANGELOG.md:10]: These two lines should not be added.

**Review by ast0815 — APPROVED** [review]

**Comment by ast0815** [comment]
@nibir98 So, did this address any of the open issues in the repository related to SAND geometry?

**Comment by ast0815** [comment]
Also, can I delete this branch now?

**Comment by nibir98** [comment]
Hi Lukas, thanks for your help. The branch can be deleted.

**Pull request merged**